src/main/kotlin/com/urielsalis/mccrashlib/deobfuscator/Deobfuscator.kt

```diff
@@ -45,37 +50,45 @@ fun getSafeChildPath(directory: File, childName: String): File? {
     return destinationFile
 }
 
-private fun getMappingFile(mappingsDirectory: File, version: String, isClient: Boolean): File? {
+private fun getMappingFile(mappingsDirectory: File, version: String, isClient: Boolean): File {
     val name = if (isClient) {
         "$version-client.txt"
     } else {
         "$version-server.txt"
     }
 
     return getSafeChildPath(mappingsDirectory, name)
+        // Sanitize `name` to avoid injection of untrusted text into exception message
+        ?: throw IllegalArgumentException("Cannot create safe path with mappings file name '${sanitize(name)}'")
 }
 
+/**
+ * @param version
+ *      Minecraft version ID
+ * @param content
+ *      Content to deobfuscate
+ * @param isClient
+ *      Whether client (`true`) or dedicated server (`false`) mappings should be used for deobfuscation
+ * @param mappingsDirectory
+ *      Existing directory where to load and store downloaded mappings
+ * @return
+ *      The deobfuscated content
+ * @throws IllegalArgumentException
+ *      If no mappings exist for the version
```

Comment:
I am not completely sure if throwing an exception here is better than returning `null`, but it at least allows us to more easily diagnose why exactly deobfuscation failed.